### PR TITLE
Setup Absinthe and add default schema

### LINKED ISSUE #59: Add Absinthe
## Expected Behavior

We will be using GraphQL in the repo. There is therefore a need to add Absinthe in the codebase as a dependency
## Current Behavior

The codebase does not have Absinthe, a GraphQL API for elixir
## Possible Solution

Add Absinthe to the deps

### PR DESCRIPTION
**Pull request by aonomike**
### What does this PR do?
- Add Absinthe, Absinthe Phoenix and  dependency 
- Make configurations for Absinthe in the supervisor, socket, endpoint ad router
- Add default schema file in the directory `~/SingForNeedsWebApp/schema/schema.ex` to enable server to run

#### Description of Task to be completed?

Adds Absinthe Phoenix to enable developers to use graphql to query the api

#### How should this be manually tested?

1. On terminal, run command `mix deps.get` then run `mix phx.server`
2. Access graphiql on your browser through url `localhost:4000/graphiql`
3. Access the api on your browser using `localhost:4000/api`

#### What are the relevant Github Issues ?

Fixes #59

### CODE REVIEW

**Comment by aonomike** [comment]
We had an issue trying to use Ecto's `Repo.insert_all/3` to add multiple records at one go and faced the issue [here](https://github.com/elixir-ecto/ecto/issues/1932). Any thoughts of the solutions attached?

File: test/sing_for_needs_web/schema/query/artists_test.exs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    conn = get conn, "/api", query: @query
+
+    assert %{"data" => %{
+      "artists" => [
+        %{"name" => "Artist 1"},
+        %{"name" => "Artist 2"},
+        %{"name" => "Artist 3"}
+      ]
+      }
+    } = json_response(conn, 200)
```
Comment (FedericoEsparza) [inline test/sing_for_needs_web/schema/query/artists_test.exs:24]: This test is insufficient and it's not catching the broken structure of your `test/support/test_helpers.ex` `artists_fixture`, that you refactored for Ebert.

Maybe look at the deleted `test/sing_for_needs_web/controllers/artist_controller_test.exs` for guidance on how to improve it or add more tests.

File: test/support/test_helpers.ex [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        name: "Artist 2",
+      }
+    Repo.insert!(artist2)
+
+    artist3 =
+      %Artist{
+        name: "Artist 3",
+      }
+    Repo.insert!(artist3)
+    [artist1, artist2, artist3]
```
Comment (FedericoEsparza) [inline test/support/test_helpers.ex:24]: For the `artists_fixture`, you're returning the `[artist1, artist2, artist3]` variables as you have them defined, not what is actually returned from `Repo.insert!`.

Each artist variable should have what's returned from the `Repo.insert!`, which, I believe, should be an `:ok` or `:error` tuple with an `%Artists{}` struct with all its new insertion information.

Do an `IO.inspect([artist1, artist2, artist3], label: "++++++")` to verify you're returning the correct version.

**Review by FedericoEsparza — COMMENTED** [review]
@aonomike @mattwr18 Great work on this PR, guys.

Please see my comments for necessary changes.

File: test/support/test_helpers.ex (hunk 2 above)
Comment (aonomike) [reply test/support/test_helpers.ex:24]: Thanks @FedericoEsparza for this. @SherSpock and I looked into these and made the changes so as to return exactly what's inserted in the database. The `insert!` method however does not return the `{:ok, %Artist{}}` tuple (is normally returned with `insert` without the bang) but just the `%Artists{}` struct with whatever value was inserted.

**Review by dhemmingson — APPROVED** [review]
We will go ahead and merge this to continue making progress. If there are any issues, we can revisit.

**Pull request merged**